fetch.bs

```diff
@@ -5914,25 +5914,55 @@ optional boolean <var>forceNewConnection</var> (default false), run these steps:
    <li><p>Return the <a for=/>appropriate network error</a> for <var>fetchParams</var>.
   </ol>
 
- <li><p>Let <var>pullAlgorithm</var> be an algorithm that <a lt=resumed for=fetch>resumes</a> the
- ongoing fetch if it is <a lt=suspend for=fetch>suspended</a>.
+ <li>
+  <p>Let |buffer| be an empty buffer that can have bytes appended to it.
+
+  <p class="note">This represents an internal buffer inside the network layer of the user agent.
+
+ <li><p>Let |pullAlgorithm| be the followings steps:
+
+  <ol>
+   <li>[=fetch/resumed|Resume=] the ongoing fetch if it is [=fetch/suspend|suspended=].
+
+   <li>Wait until |buffer| is not empty.
+
+   <li>Let |available| be the size of |buffer|.
+
+   <li>Let |desiredSize| be |available|.
+
+   <li>If |stream|'s [=ReadableStream/current BYOB request view=] is non-null, then set
+   |desiredSize| to |stream|'s [=ReadableStream/current BYOB request view=]'s [=BufferSource/byte
+   length=].
+
+   <li>Let |extractSize| be the smaller value of |available| and |desiredSize|.
+
+   <li>Let <var>bytes</var> be the result of extracting |extractSize| of bytes from
+   <var>buffer</var>.
+
+   <li>If |stream|s [=ReadableStream/current BYOB request view=] is non-null, then
+   [=ArrayBufferView/write=] |bytes| into |stream|'s [=ReadableStream/current BYOB request view=],
+   and set |view| to |stream|'s [=ReadableStream/current BYOB request view=]. <li>Otherwise, set
+   |view| to the result of [=ArrayBufferView/create|creating=] a {{Uint8Array}} from |bytes| in
+   |stream|'s [=relevant Realm=].
+
+   <li>[=ReadableStream/Enqueue=] |view| into |stream|.
```

Comment:
Don't enqueue the current BYOB request view. Only enqueue if you needed to create a new view.

Reply:
Isn't it something recommended by Streams? Below https://streams.spec.whatwg.org/#readablestream-current-byob-request-view:

>Specifications should, when possible, [write](https://webidl.spec.whatwg.org/#arraybufferview-write) into the [current BYOB request view](https://streams.spec.whatwg.org/#readablestream-current-byob-request-view) when it is non-null, and then call [enqueue](https://streams.spec.whatwg.org/#readablestream-enqueue) with that view.

It doesn't have a separate algorithm for `respond`, so I think enqueuing is the only way to ping the stream...

Reply:
You're right, I totally forgot that we wrapped up the JS `enqueue()` + `respond()` into a single spec-level call. Sorry!